src/routes/blog/post/remix-3-whats-changing-and-why-it-matters/+page.markdoc

```diff
@@ -0,0 +1,144 @@
+
+When a loader throws a `Response` with an error status, Remix automatically triggers the matching error boundary. For example, if a loader throws `new Response("Not found", { status: 404 })`, Remix will show the Not Found UI for that route but keep parent layouts mounted.
+
+This approach prevents a single broken component from blanking the entire page (a common problem in many SPAs) and helps developers handle network or data errors cleanly.
+
+### TypeScript and developer experience
+
+Remix 3 is written in TypeScript from the start, so every route, loader, and action is strongly typed. Route parameters and return types are inferred automatically. For example, if `routes.books.show` has a `:slug` parameter, TypeScript will know that `params.slug` is a string inside its handlers.
+
+Because Remix avoids code generation, you don't need a separate typegen step. The framework infers everything from the code you already write. The same runtime-first philosophy also means you can run your app directly in `ts-node` during development, without waiting for a full build.
+
+The development workflow is simpler, too. The preview demos show a single `remix dev` command that starts a local server, watches files, and reloads automatically, no webpack config or complex bundler setup required.
+
+### Deployment and environment support
+
+Remix 3's platform independence is one of its biggest selling points. Since it's built on the Fetch API, the same app can run on Node, Deno, Bun, or Cloudflare Workers with almost no changes.
+
+The router handles incoming requests as `Request` objects and produces `Response` objects, so deployment often comes down to a few lines:
+
+```ts
+import { createRequestListener } from "@remix-run/serve";
+import { router } from "./router";
+
+export default createRequestListener({ router });
+```
+
+That pattern works in Node, Vercel, Netlify, or Fly.io. On Cloudflare, it's even simpler, just export `{ fetch: createFetchHandler(router) }`. Because the framework doesn't rely on Node-specific APIs, there's nothing to rewrite when switching platforms.
+
+This also makes Remix 3 easier to scale. You can deploy it as serverless functions, edge workers, or a long-running Node process depending on your needs. The framework doesn't force a specific host or vendor. That independence is deliberate; the team wants Remix 3 apps to run anywhere, without lock-in.
+
+### Trade-offs and what to expect
+
+Rebuilding a framework from scratch isn't free. Dropping React means losing direct access to its huge ecosystem. Libraries like Material UI or React Query won't plug in directly. Developers will either use plain web components, small JS libraries, or future Remix-specific packages.
+
+There's also a learning curve. Without React's hooks or automatic rendering, you'll manage updates yourself. That's simpler in concept but different in habit. It's a shift from "declare and forget" to "update when needed."
```

Comment:
Come to think of it, do we have contact at Remix? they might want to review ot too, for technical details acuracy.